pontoon/base/placeables.py

```diff
@@ -3,38 +3,32 @@
 
 class PythonFormatNamedString:
     parser = re.compile(
-        r"(%\([[\w\d!.,[\]%:$<>+\-= ]*\)[+|-|0\d+|#]?[.\d+]?[s|d|e|f|g|o|x|c|%])",
+        r"(%\([[\w\d!.,[\]%:$<>+\-= ]*\)[+-0\d+#]?[.\d+]?[sdefgoxc%])",
         re.IGNORECASE,
```

Comment:
Presuming that this is attempting to capture [printf-style strings](https://docs.python.org/3/library/stdtypes.html#printf-style-string-formatting), it doesn't really make sense to have both this and `PythonFormattingVariable`; really both should be merged into one with a rewritten regex like this, and call it something like `PythonPrintfString`.

```suggestion
        r"""
            %
            (\(.*?\))?  # mapping key
            [\#0\-\ +]* # conversion flags
            [\d*]*      # minimum field width
            (\.[\d*])?  # precision
            [hlL]?      # length modifier
            [acdEeFfGgiorsuXx%]  # conversion type
        """,
        re.X,
```

Reply:
`test_PythonFormatNamedString()` and `test_PythonFormattingVariable()` are both failing with this change applied:

```
AssertionError: assert [('(name)', '')] == ['%(name)s']
AssertionError: assert [('(name)', '')] == ['%(name)d']
```

Reply:
I updated the suggestion in the above comment to remove the capturing groups (i.e. it's now using `(?: ... )` rather than `( ... )`), which should fix the issue.